fix: Allow download url in video component when MP4/WEBM format is available

sathiscode opened the pull request:
## Description
When creating a new video component in Studio and specifying an HLS URL and MP4 URL along with setting Video Download Allowed equal to True the download link for MP4 has to be shown.

For example, if we have set the URL as below and set Video Download to True,
![image](https://user-images.githubusercontent.com/1616291/226628136-bad8e6bd-1443-4cf3-97b9-42265de36fd0.png)

Then Before:
![image](https://user-images.githubusercontent.com/1616291/226627774-e6569731-deec-41c3-8a11-ba09f22134b9.png)

After:
![image](https://user-images.githubusercontent.com/1616291/226627887-10492688-58de-4ec4-bae8-5fa3e0ee570e.png)

## Supporting information
Resolves https://github.com/openedx/edx-platform/issues/31300

## Deadline
"None"

Private-ref: [BB-7249](https://tasks.opencraft.com/browse/BB-7249)

openedx-webhooks commented:
Thanks for the pull request, @sathiscode!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

mphilbrick211 commented:
Hi @sathiscode! Thank you for this contribution. In order to proceed, please submit a CLA form as mentioned above. Let me know if you have any questions. Thanks!

sathiscode commented:
> Hi @sathiscode! Thank you for this contribution. In order to proceed, please submit a CLA form as mentioned above. Let me know if you have any questions. Thanks!

I've signed the CLA. Looks like it will take some time to reflect here.

tecoholic commented on xmodule/video_block/video_block.py:
```diff
@@ -357,11 +357,8 @@ def get_html(self, view=STUDENT_VIEW, context=None):  # lint-amnesty, pylint: di
-            if download_video_link and download_video_link.endswith('.m3u8'):
-                download_video_link = None
+                # If there are multiple html5 sources, we use the first non HLS video urls
+                download_video_link = next(filter(lambda vid: not vid.endswith('.m3u8'), self.html5_sources), None)
```
```suggestion
                download_video_link = next((url for url in self.html5_sources if not url.endswith('.m3u8')),  None)
```

tecoholic replied:
@sathiscode I think filter+lambda is probably a overkill for a simple if check. Do you think we can simplify it as I have suggested above?

sathiscode replied:
Yes, certainly your solution is much simpler. Will change it.

tecoholic commented on lms/djangoapps/courseware/tests/test_video_mongo.py:
```diff
@@ -1129,6 +1129,23 @@ def test_get_html_hls_no_video_id(self):
         context = self.item_descriptor.render(STUDENT_VIEW).content
         assert "'download_video_link': None" in context
 
+
```
@sathiscode A double empty line here will probably cause the linter check to fail during the CI. Can you kindly check and remove the extra line if present?

tecoholic commented:
@sathiscode Thank you for the updating the PR. Can you squash all your commits and write a commit message following the [conventional commits](https://github.com/openedx/open-edx-proposals/blob/master/oeps/best-practices/oep-0051-bp-conventional-commits.rst) format? Kindly include `Fixes: https://github.com/openedx/edx-platform/issues/31300` in the footer of the commit message.

tecoholic commented:
@sathiscode :+1:

- [x] I tested this: Verified that the Download link is shown.
- [x] I read through the code
- [x] I checked for accessibility issues
- [ ] Includes documentation - NA

e0d commented:
Updating the branch with a merge commit and running the tests.

e0d commented:
@sathiscode it looks like there are some linter/quality failures, can you have a look?

sathiscode committed:
fix: Allow download url in video component when MP4/WEBM format is available

When creating a new video component in Studio and specifying an HLS URL and MP4 URL along with setting Video Download Allowed equal to True the download link for MP4 has to be shown.

Fixes: https://github.com/openedx/edx-platform/issues/31300

sathiscode commented:
> @sathiscode it looks like there are some linter/quality failures, can you have a look?

@e0d linter issue is fixed. Can you please check.

Agrendalath reviewed (approved):
:+1:

* [x] I tested this: checked the video download link
* [x] I read through the code
* [x] I checked for accessibility issues: n/a
* [x] Includes documentation: n/a
* [x] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository: n/a

openedx-webhooks commented:
@sathiscode 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/31971&entry.1484655318=https://github.com/sathiscode&entry.752974735=2023-03-21+13:57&entry.1917517419=2023-04-07+15:47&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
